pages/advanced/erc-7579/tutorials/7579-tutorial.mdx

```diff
@@ -0,0 +1,194 @@
+
+- Downloaded and installed [Node.js](https://nodejs.org/en/download/package-manager) and [pnpm](https://pnpm.io/installation).
+- Created an API key from [Pimlico](https://www.pimlico.io/).
+
+**Note:** If you wish to follow along using the completed project, you can [check out the GitHub repository](https://github.com/5afe/safe-7579-tutorial) for this tutorial.
+
+## **1. Setup a Next.js application**
+
+Initialize a new Next.js app using pnpm with the following command:
+
+```shell
+pnpm create next-app
+```
+
+When prompted by the CLI:
+
+- Select `yes` to TypeScript, ESLint, and App router.
+- Select `no` to all other questions (Tailwind, `src` directory, and import aliases).
+
+### **Install dependencies**
+
+For this project, we'll use Pimlico's [Permissionless.js](https://docs.pimlico.io/permissionless) to set up a Safe, Rhinestone's [Module SDK](https://docs.rhinestone.wtf/module-sdk) for ERC-7579 functionalities, and [viem](https://www.npmjs.com/package/viem) for some helper functions.
+
+Run the following command to add all these dependencies to the project:
+
+```shell
+pnpm add permissionless viem @rhinestone/module-sdk
+```
+
+Now, create a file named `.env.local` at the root of your project, and add your Pimlico API key to it:
+
+```shell
+echo "NEXT_PUBLIC_PIMLICO_API_KEY='your_pimlico_api_key_goes_here'" > .env.local
+```
+
+### **Run the development server**
+
+Run the local development server with the following command:
+
+```shell
+pnpm dev
+```
+
+Go to `http://localhost:3000` in your browser to see the default Next.js application.
+
+![screenshot-default](../../../../assets/next.png)
+
+## 2. Initialize `permissionless` client
+
+Create a `lib` folder at project root, and add a file `permissionless.ts`:
+
+```shell
+mkdir lib
+cd lib
+touch permissionless.ts
+```
+
+Add the code necessary to create Pimlico’s `smartAccountClient` by adding this content to `permissionless.ts` :
+
+```typescript
+// from ../../../../examples/erc-7579/lib/permissionless.ts
+```
+
+It will load a `PRIVATE_KEY` from `.env.local` (or generate one if it doesn’t exist) and then create the `publicClient`, `paymasterClient` and `bundlerClient` necessary to initialize the `permissionless` library. It then creates a 7579-enabled Safe Smart Account from the generated private key, before passing it to `createSmartAccountClient` to generate a Permissionless client.
+
+## 3. Add Rhinestone module functionality
+
+Create a new file `scheduledTransfers.ts` in the `lib` folder:
+
+```shell
+touch scheduledTransfers.ts
+```
+
+Add the code necessary to create a scheduled transfer using Rhinestone’s `ScheduledTransfers` module:
+```typescript
+// from ../../../../examples/erc-7579/lib/scheduledTransfers.ts
+```
+
+This file contains two functions:
+- `install7579Module` will install the module to a Safe and schedule its first transfer; 
+- `scheduleTransfer` to schedule transfers in a Safe where the module has been previously installed.
+
```

Comment:
Would be good to mention what other features this modules provides to give some ideas about the power of this module. We can mention things like recurring transfers, etc. Also to mention that we've skipped over some of these to keep the tutorial brief, but they can try these out.